partiql-eval/src/pattern_match.rs

```diff
@@ -0,0 +1,150 @@
+// TODO docs
+
+// TODO consider how to use the appropriate mechanisms to prevent exhaustion of
```

Comment:
You reckon as a short term measure we can put a hard limit for now?

Reply:
added a limit of `const RE_SIZE_LIMIT: usize = 1 << 16;` for now. It's a bit arbitrarily chosen, as I don't have a good intuition for what limit corresponds to. 

Cf. https://docs.rs/regex/latest/regex/struct.RegexBuilder.html#method.size_limit